### PR TITLE
Make test case slightly more robust.

### PR DESCRIPTION
**Pull request by adrian-prantl**
Reported in rdar://33876675 , rdar://33993056.

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test and merge

**Comment by adrian-prantl** [comment]
@swift-ci test and merge

File: test/DebugInfo/modulecache.swift [hunk 1]
```diff
@@ -7,9 +7,9 @@ import ClangModule
 
 // 1. Test that swift-ide-test creates a thin module without debug info.
 
+// RUN: %empty-directory(%t)
 // RUN: %swift-ide-test_plain -print-usrs -target %target-triple -module-cache-path %t  -I %S/Inputs -source-filename %s
-// RUN: %target-swift-frontend %s -c -g -o %t.o -module-cache-path %t -I %S/Inputs
-// RUN: file %t/*/ClangModule-*.pcm | grep data
+// RUN: xxd %t/*/ClangModule-*.pcm | head -n 1 | grep -q CPCH
```
Comment (gottesmm) [inline test/DebugInfo/modulecache.swift]: Is xxd available on all platforms? Maybe this should be conditionalized?

**Comment by adrian-prantl** [comment]
I tested (Ubuntu) Linux and Darwin. I really don't know about SystemZ.

**Comment by gottesmm** [comment]
@adrian-prantl my suggestions: Just do a which check in the lit.cfg file and add a feature. If the feature isn't set, skip the test.

**Comment by gottesmm** [comment]
Then there are no worries.

**Comment by adrian-prantl** [comment]
I got rid of the xxd dependency by using only `head -c`.

**Comment by adrian-prantl** [comment]
@swift-ci please test and merge

**Comment by adrian-prantl** [comment]
@swift-ci Please test and merge

**Pull request merged**